opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -854,7 +853,17 @@ namespace Opm {
         }
     }
 
-    // static member function
+    
+    const MessageContainer& Eclipse3DProperties::getMessageContainer() const {
```

Comment:
If ``GridPropertie`` does not have ``MessageContainer`` itself, some methods both in ``GridProperties`` and ``Eclipse3DProperties`` need to be modified to accept the ``MessageContainer`` reference from ``Eclipse3DProperties``, like ``initPORV``, ``addKeyword``, ``getOrCreateProperty``, etc.  A quick ``grep`` shows some of these funcitons are not used in other opm module, that shows we don't need to modify the other calling scope, but I'm not sure about this.